lib/aur-sync

```diff
@@ -386,6 +390,28 @@ else
     xargs -a "$tmp"/queue stat >/dev/null || exit 2 # ensure all directories are available
 fi
 
+if (( auto_key_retrieve )); then
+    # shellcheck disable=SC2174
+    mkdir -pm 0700 -- "$AUR_SYNC_GNUPGHOME"
+    declare -A keys_uniq
+
+    # Retrieve unique set of gpg keys to be imported
+    while IFS= read -r path; do
```

Comment:
Mostly because this isn't nul delimited but perhaps a good idea anyway to check if `"$path"/.SRCINFO` exists before attempting to use it.

``` bash
while IFS= read -r path; do
    if [[ -f $path/.SRCINFO ]]; then
        mapfile -t < <(pacini "$path"/.SRCINFO 'validpgpkeys')

        for key in "${MAPFILE[@]}"; do
            keys_uniq[$key]=1
        done
    fi
done < "$tmp"/queue
```